### PR TITLE
[Gradient Compression] Add register_comm_hook API to DDP communication hooks documentation page

### PR DESCRIPTION
**Pull request by wayi1**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51846 [Gradient Compression] Add register_comm_hook API to DDP communication hooks documentation page**

`register_comm_hook` method is defined in DistributedDataParallel module, but it is not covered in `distributed.rst`. Since it's closely related to DDP communication hook, add the docstrings to `ddp_comm_hooks.rst` instead of a reference.

Screenshot:
Screen Shot 2021-02-06 at 4 17 10 PM

Differential Revision: [D26298191](https://our.internmc.facebook.com/intern/diff/D26298191/)

### CODE REVIEW

**Review by rohan-varma — APPROVED** [review]
LGTM, although I think the doc build is failing, please take a look at that:
```
Feb 08 06:37:28 waiting for workers...
Feb 08 06:37:45 /opt/conda/lib/python3.6/site-packages/torch/distributed/distributed_c10d.py:144: UserWarning: torch.distributed.reduce_op is deprecated, please use torch.distributed.ReduceOp instead
Feb 08 06:37:45   warnings.warn("torch.distributed.reduce_op is deprecated, please use "
Feb 08 06:38:10 /var/lib/jenkins/workspace/docs/source/generated/torch.nn.parallel.DistributedDataParallel.rst:358: WARNING: duplicate object description of torch.nn.parallel.DistributedDataParallel.register_comm_hook, other instance in ddp_comm_hooks, use :noindex: for one of them
Feb 08 06:38:10 looking for now-outdated files... none found
Feb 08 06:38:10 pickling environment... done
Feb 08 06:38:10 checking consistency... done
Feb 08 06:38:10 running tests...
Feb 08 06:38:11 
```
